### ISSUE TITLE
Creation of `cas_view_only` user role

### ISSUE THREAD

**Issue by patrickisaac** [bcgov/cas-registration#2494]
#### Dependencies

Blocked by #2493 (not technically blocked, but will avoid conflicts and be easier to wrap our heads around)
Same with #2522 

#### Description:
As a `cas_view_only` user, I should have `view-only` functionality as listed in this [table](https://miro.com/app/board/uXjVKdOp2dg=/?share_link_id=560390359720). PR started here: https://github.com/bcgov/cas-registration/pull/2521/files#diff-ec9c317c0857a29d2d59d94f0a0bc81f91a3b6dd731246575bcbb206c9d41161. The BE permissions will probably need to be redone, but the dashboard and FE routing should be reusable.

#### Acceptance Criteria:
- [ ] View-only

#### Development Checklist:

- [x] migration to add the cas_view_only role (e.g. of adding cas_analyst: cas-registration/bc_obps/registration/migrations/0002_prod_data.py, or see how cas_director is added)
- [x] in cas-registration/bciers/apps/registration/app/idir, create a cas_view_only folder. Copy/paste the structure of cas_admin into this folder (cas_view_only is allowed to view everything)
- [x] add cas_view_only to the "access_roles" in the internal.json files to give cas_view_only users access to all dashboard tiles
~- [ ] add conditionality to the operator admin role approvals so that cas_view_only cannot manage first-time operator admin role approvals (only cas_director and cas_analyst can). Could fetch the role in the grid component, or pass the role_name (we know what it is from the file structure) as a prop in page.tsx~ approval feature isn't built yet; added this to #2059 instead
- [x] can't assign boro or bcghg id (this should already be done as part of #2493 
- [x] Check and update the endpoint and service permissions for this new role to ensure the user has the correct access.
- [ ] [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

**Comment by andrea-williams** [comment]
I recommend (with agreement from some other devs) that it would be better to name this role `cas_read`, `cas_view_only`, `cas_read_only`, or the like, for the sake of being more explicit in what that role does. `cas_user` is very generic.

**Comment by patriciarussellCAS** [comment]
Let's change it! I vote to change it to cas_view_only
@andrea-williams

**Comment by vesselak** [comment]
assumed cas_view-only and navigated back and forth to confirm the role can view but not perform an action. accepting ticket.
